extensions/positron-r/src/provider.ts

```diff
@@ -399,35 +397,24 @@ async function findCurrentRBinaryFromRegistry(): Promise<string | undefined> {
  * @returns The install path for R, or undefined if an R installation file could
  * not be found at the install path.
  */
-async function getRegistryInstallPath(hive: string, wow?: string | undefined): Promise<string | undefined> {
-	try {
-		const key = new winreg({
-			hive: hive as keyof typeof winreg,
-			// 'R64' here is another place where we explicitly ignore 32-bit R
-			// Amend a WOW path after "Software" if requested
-			key: `\\Software\\${wow ? wow + '\\' : ''}R-Core\\R64`,
-		});
+async function getRegistryInstallPath(hive: 'HKEY_CURRENT_USER' | 'HKEY_LOCAL_MACHINE', wow?: string | undefined): Promise<string | undefined> {
+	// 'R64' here is another place where we explicitly ignore 32-bit R
+	// Amend a WOW path after "Software" if requested
+	const R64_KEY: string = `SOFTWARE\\${wow ? wow + '\\' : ''}R-core\\R64`;
 
-		LOGGER.info(`Checking for 'InstallPath' in registry key ${key.key} for hive ${key.hive}`);
-
-		const result = await new Promise<{ value: string }>((resolve, reject) => {
-			key.get('InstallPath', (error, result) => {
-				if (error) {
-					reject(error);
-				} else {
-					resolve(result);
-				}
-			});
-		});
+	if (os.platform() !== 'win32') {
+		LOGGER.info('Skipping registry check on non-Windows platform');
+		return undefined;
+	}
```

Comment:
I'll probably move this up in the next PR, where I truly take more advantage of the registry. This is all about to get refactored/built upon.